src/config/config.go

```diff
@@ -202,10 +206,97 @@ func LoadConfig(path string, filterByOS bool) error {
 	return nil
 }
 
+// FillActiveTemplate handles setting the active variables and reloading the base template.
+func FillActiveTemplate() error {
+	packageVariables, err := utils.FindYamlTemplates(&active, "###ZARF_PKG_VAR_", "###")
+	if err != nil {
+		return err
+	}
+
+	for key := range CommonOptions.SetVariables {
+		value := CommonOptions.SetVariables[key]
+		packageVariables[key] = &value
+	}
+
+	for key, value := range packageVariables {
+		if value == nil {
+			if !CommonOptions.Confirm {
+				if setVal, err := promptVariable(key, nil); err == nil {
+					packageVariables[key] = &setVal
+				} else {
+					return err
+				}
+			} else {
+				return fmt.Errorf("variable '%s' must be '--set' when using the '--confirm' flag", key)
+			}
+		}
+	}
+
+	templateMap := map[string]string{}
+	for key, value := range packageVariables {
+		// Variable keys are always uppercase in the format ###ZARF_PKG_VAR_KEY###
+		templateMap[strings.ToUpper(fmt.Sprintf("###ZARF_PKG_VAR_%s###", key))] = *value
+	}
+
+	return utils.ReloadYamlTemplate(&active, templateMap)
+}
+
+// SetActiveVariables handles setting the active variables used to template component files.
+func SetActiveVariables() error {
```

Comment:
Can we make a new `variables.go` or something in the `config` pkg for all this logic?  It's getting a little messy.

Reply:
Created a new `variables.go` file and flattened the nested if statements to hopefully make this more readable.